app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java

```diff
@@ -0,0 +1,205 @@
+            Log.e("ResultActivity", "error retrieving last photo");
+        }
+
+        AsyncLoad ocrTask = new AsyncLoad(mOCRTextView, getString(R.string.processing));
+        ocrTask.execute(lastPhoto);
+    }
+
+    /**
+     * Menu inflater
+     * @author Francesco Pham
+     */
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.result_menu, menu);
+        return true;
+    }
+
+    /**
+     * Handling click events on the menu
+     * @author Francesco Pham - modified by Stefano Romanello
+     */
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle item selection
+        switch (item.getItemId()) {
+            case R.id.test:
+                Intent i = new Intent(ResultActivity.this, TestResultActivity.class);
+                startActivity(i);
+                return true;
+            case R.id.download_photos:
+                Intent download_intent = new Intent(ResultActivity.this, DownloadDbActivity.class);
+                startActivity(download_intent);
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    /**
+     * Execute a task and post the result on the TextView given on construction
+     * (g3) - modified by Rossi Leonardo - modified by Andrea Ton (barcode)
+     */
+    @SuppressLint("StaticFieldLeak")
+    private class AsyncLoad extends AsyncTask<Bitmap, Void, String> {
+
+        private ProgressDialog progressDialog;
+        private TextView resultTextView;
+        private String progressMessage;
+
+        AsyncLoad(TextView view, String progressMessage) {
+            this.resultTextView = view;
+            this.progressMessage = progressMessage;
+        }
+
+        @Override
+        protected String doInBackground(Bitmap... bitmaps) {
+            TextExtractor ocr = new TextExtractor();
+            String textRecognized = "";
+            String barcodeRecognized = "";
+
+            if(lastPhoto != null) {
+                //invoke a barcodeRecognizer choosing the api type
+                Barcode barcodeRecognizer = barcodeRecognizer(BarcodeRecognizer.API.mlkit);
+                //get the String of the barcode from the bitmap image
+                barcodeRecognized = barcodeRecognizer.decodeBarcode(lastPhoto);
+                //check if any barcode is found
+                if(barcodeRecognized.equals(""))
+                {
+                    //if no barcode was found than try to get the text ocr
+                    textRecognized = ocr.getTextFromImg(lastPhoto);
+
+                    if(textRecognized.equals(""))
+                    {
+                    //if nothing was found than return a No text found message
+                    textRecognized = getString(R.string.no_text_found);
+                    final String finalTextRecognized = textRecognized;
```

Comment:
not understand why you created a final at this point and don't just leave the finalTextRecognized as finale variabiles

Reply:
Inner classes in Java only capture variables that are declared final.